### ISSUE TITLE
YouTube

### ISSUE THREAD

**Issue by UltimateRiff** [kellnerd/harmony#31]
continuing from discussion [here](https://github.com/kellnerd/harmony/issues/5#issuecomment-2147779359).

so, after a very brief search, it seems there's no official YouTube Music API, only one for YouTube (and a few unofficial ones for YouTube Music)

- https://developers.google.com/youtube/v3/docs

a few items to be aware of specific to YouTube with examples where applicable:
- metadata in the description is not at all standardized, perhaps save for distributed content (i.e. from distributors like DistroKid). this is probably only an issue if we want to eventually add relationships with Harmony (once that's possible of course)
  - (will add examples of distributed content soon™)
  - https://www.youtube.com/watch?v=s742C0v5SFI (unstandardized with no heading)
  - https://www.youtube.com/watch?v=Fv3wm5CFD5g (unstandardized with implied section)
- a single video can have different titles, descriptions, and possibly different audio tracks per region
  - https://www.youtube.com/watch?v=NT6Pf28eCgQ
  - https://www.youtube.com/watch?v=4Hg0Nchl3HU
- often an album will be released as a single video with chapter markers for each track (I believe these come from the video description)
  - https://www.youtube.com/watch?v=k9ynZnEBtvw
  - https://www.youtube.com/watch?v=6A4Lu6XAlh0 (example with improperly formatted chapter markers)
- video titles are not standardized, sometimes containing the artist name or [official video] or other nonsense, which maybe should or shouldn't be removed from MusicBrainz submissions? could probably add release disambiguations based on these (such as Visualizer, Lyric Video, Music Video, etc.)
- there are categories for YouTube videos (including Music and Entertainment, to name a couple). I don't know if those would be important to use, as some "music" releases might be non-music (i.e. about Music Production or about Music), and some music videos might not be categorized as such (as well as some podcasts and other items people might want to import). I don't think these categories are visible on the video pages, but might be in the API

**Comment by UltimateRiff** [comment]
also, on the note of distributed content on YouTube, I feel like those should probably get YouTube Music links instead of YouTube links (the only change is YouTube Music has the music subdomain, I believe)

**Comment by UltimateRiff** [comment]
I was also told to mention [YouTube.js](https://github.com/LuanRT/YouTube.js) over here too

**Comment by maximschwalm** [comment]
> https://www.youtube.com/watch?v=i5kd-3954x8 (typical DistroKid release with few credits)
> https://www.youtube.com/watch?v=9De6UsSY-bo (higher quality medadata, with production and composition credits)

@UltimateRiff These aren't traditional YouTube videos but so-called "Art Tracks".
I've found some information about them here:

- https://support.google.com/youtube/answer/6007071?hl=en&ref_topic=3504274
- https://verseone.net/answer/youtube-art-track-everything-you-need-to-know/

**Comment by UltimateRiff** [comment]
right, but I think they are treated just like regular YouTube videos by at least the web interface (and likely the API too? not 100% on that)

**Comment by UltimateRiff** [comment]
right, but I think they are treated just like regular YouTube videos by at least the web interface (and likely the API too? not 100% on that)

that said, usually these tracks are part of an album release on YouTube, so that would be something to look for (tho singles are possible too)
